src/test/java/uk/ac/ebi/atlas/hcalandingpage/HcaHumanExperimentServiceTest.java

```diff
@@ -27,37 +27,50 @@
-                .thenReturn(ImmutableSet.of(EXPERIMENT_ACCESSION));
-        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", "skin"))
+        subject = new HcaHumanExperimentService(experimentTraderMock, hcaHumanExperimentDaoMock);
+    }
+
+    @Test
+    public void ShouldGetAllHumanExperimentsWithEmptyCharacteristicValue() {
+        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", ImmutableSet.of()))
                 .thenReturn(ImmutableSet.of(EXPERIMENT_ACCESSION));
-        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", "foo"))
-                .thenReturn(ImmutableSet.of());
+        assertThat(subject.getPublicHumanExperiments("organism_part", ImmutableSet.of()))
+                .isNotEmpty().hasSize(1);
+    }
 
-        subject = new HcaHumanExperimentService(experimentTraderMock, hcaHumanExperimentDaoMock);
+    @Test
+    public void ShouldGetHumanExperimentsOnlyForTheGivenCharacteristicValue() {
+        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", ImmutableSet.of("skin")))
+                .thenReturn(ImmutableSet.of(EXPERIMENT_ACCESSION));
+        assertThat(subject.getPublicHumanExperiments("organism_part", ImmutableSet.of("skin")))
+                .isNotEmpty().hasSize(1);
     }
 
     @Test
-    public void sizeIsRightForWithDefaultValueOfParameter() {
-        assertThat(
-                subject.getPublicHumanExperiments("organism_part", ""))
-                .hasSize(1);
+    public void shouldGetEmptyExperimentsIfTheGivenCharacteristicValueIsInvalid() {
+        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", ImmutableSet.of("foo")))
+                .thenReturn(ImmutableSet.of());
+        assertThat(subject.getPublicHumanExperiments("organism_part", ImmutableSet.of("foo")))
+                .isEmpty();
     }
 
     @Test
-    public void sizeIsRightForCorrectCharacteristicNameAndCharacteristicValue() {
-        assertThat(
-                subject.getPublicHumanExperiments("organism_part","skin"))
-                .hasSize(1);
+    public void shouldGetHumanExperimentsForMultipleCharacteristicValues() {
+        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("organism_part", ImmutableSet.of("skin", "lymph node")))
+                .thenReturn(ImmutableSet.of(EXPERIMENT_ACCESSION));
+        assertThat(subject.getPublicHumanExperiments("organism_part", ImmutableSet.of("skin", "lymph node")))
+                .isNotEmpty().hasSize(1);
     }
 
     @Test
-    public void noExperimentReturnedForInvalidCharacteristicValue() {
-        assertThat(
-                subject.getPublicHumanExperiments("organism_part","foo"))
-                .hasSize(0);
+    public void ShouldGetAllHumanExperimentsWithEmptyCharacteristicNameAndValue() {
+        when(hcaHumanExperimentDaoMock.fetchExperimentAccessions("", ImmutableSet.of()))
```

Comment:
If you move `when(...) then(...)` statements in the `setUp` method then you would not need to repeat this statement in multiple methods also the test case themselves would be much cleaner and simple.

Reply:
It's true if we are using the same mocking in multiple test cases. 
But we are not repeating mock scenarios, they are different from each other.
If we move them back to setUp(), all mock scenarios would execute for every test case. 
Let me know if you are not happy with this, I can move them back to setUP() as you're the owner of this code :).

Reply:
Ohh I missed it earlier you are right they are different for each test so makes sense 👍 
Regarding ownership of the code, we all are owners of it as a team 😄